py/redrock/rebin.py

```diff
@@ -98,14 +181,40 @@ def trapz_rebin(x, y, xnew=None, edges=None):
         >>> y = np.ones(10)
         >>> trapz_rebin(x, y, edges=[0,2,4,6,8])  #- density still 1, not 2
         array([ 1.,  1.,  1.,  1.])
+        >>> y = np.ones((2,10)) #nbasis = 2
+        >>> y[1,:] = np.arange(10)
+        >>> rebin.trapz_rebin(x, y, edges=[0,2,4,6,8], use_gpu=True) #nbasis=2, GPU mode
+        array([[1., 1.],
+               [1., 3.],
+               [1., 5.],
+               [1., 7.]])
+        >>> rebin.trapz_rebin(x, y, edges=[0,2,4,6,8], myz=[0,0.5],use_gpu=True) #nbasis=2, multiple redshifts, GPU mode
+        array([[[1. , 1. ],
+                [1. , 3. ],
+                [1. , 5. ],
+                [1. , 7. ]],
+               [[1. , 0.5],
+                [1. , 1.5],
+                [1. , 2.5],
+                [1. , 3.5]]])
 
     Args:
         x (array): input x values.
-        y (array): input y values.
+        y (1-d or 2-d array): input y values (GPU mode allows 2-d array
+            with multiple bases). 
         edges (array): (optional) new bin edges.
+        myz (array): (optional) redshift array to rebin in batch on GPU,
+            applying redshifts on-the-fly to x
+        use_gpu (boolean): whether or not to use GPU batch mode 
 
     Returns:
         array: integrated results with len(results) = len(edges)-1
+            In GPU batch mode, returns cp.array with shape (nz, nbin, nbasis)
+            where nbin = len(results) = len(edges)-1
+            if nz or nbasis is 1, this dimension will be squeezed,
+            e.g. for nbasis = 1 and nz = 100, the shape will be (100, nbin)
+            while for nbasis = 1 and nz = 1 the shape will be (nbin), the
+            same as CPU mode.
```

Comment:
This is done in the new version with the exception of the order of dimensions, which remains (nz, nbins, nbasis) as discussed because that is what is expected by zscan.